cabal.project

```diff
@@ -10,12 +10,15 @@ test-show-details: direct
 
 benchmarks: True
 
+package *
+  ghc-options: -Wunused-packages
```

Comment:
Hmm. I think we normally put warning options in the cabal files, can we do that here too? Putting them all in `cabal.project` wouldn't be totally unreasonable, it's just inconsistent.

Reply:
I was just lazy :smile: 
But I agree and I moved it to cabal files + slightly consolidating using `common warnings` stanza and fixed few more warnings.
I hope there's nothing controversial, but will wait few days for your feedback (or let you merge if you think it's ok).